chain-cli/chaincode-template/src/apples/AppleContract.spec.ts

```diff
@@ -21,57 +21,60 @@ import {
   writesMap
 } from "@gala-chain/test";
 
-import { AppleTree, AppleTreeDto, PickAppleDto, Variety } from "../apples";
+import { AppleTree, PickAppleDto, Variety } from "../apples";
 import { AppleContract } from "./AppleContract";
+import { PlantAppleTreeDto } from "./dtos";
 
 it("should allow to plant a tree", async () => {
   // Given
   const user = users.random();
-  const { contract, ctx, writes } = fixture(AppleContract).registeredUsers(user);
-  const dto = new AppleTreeDto(Variety.GALA, 1).signed(user.privateKey);
+  const { contract, ctx, getWrites } = fixture(AppleContract).registeredUsers(user);
+  const dto = new PlantAppleTreeDto(Variety.GALA, 1, randomUniqueKey()).signed(user.privateKey);
   const expectedTree = new AppleTree(user.identityKey, dto.variety, dto.index, ctx.txUnixTime);
 
   // When
   const response = await contract.PlantTree(ctx, dto);
 
   // Then
   expect(response).toEqual(transactionSuccess());
-  expect(writes).toEqual(writesMap(expectedTree));
+  expect(getWrites()).toEqual(writesMap(expectedTree));
 });
 
 it("should fail to plant a tree if tree already exists", async () => {
   // Given
   const user = users.random();
 
-  const { contract, ctx, writes } = fixture(AppleContract)
+  const { contract, ctx, getWrites } = fixture(AppleContract)
     .registeredUsers(user)
     .savedState(new AppleTree(user.identityKey, Variety.GOLDEN_DELICIOUS, 1, 0));
 
-  const dto = new AppleTreeDto(Variety.GOLDEN_DELICIOUS, 1).signed(user.privateKey);
+  const dto = new PlantAppleTreeDto(Variety.GOLDEN_DELICIOUS, 1, randomUniqueKey()).signed(user.privateKey);
 
   // When
   const response = await contract.PlantTree(ctx, dto);
 
   // Then
   expect(response).toEqual(transactionErrorMessageContains("Tree already exists"));
-  expect(writes).toEqual({});
+  expect(getWrites()).toEqual({});
 });
 
 it("should allow to pick apples", async () => {
   // Given
   const twoYearsAgo = new Date(new Date().getTime() - 1000 * 60 * 60 * 24 * 365 * 2).getTime();
   const tree = new AppleTree("client|some-user", Variety.GALA, 1, twoYearsAgo);
   const user = users.random();
-  const { contract, ctx, writes } = fixture(AppleContract).registeredUsers(user).savedState(tree);
+  const { contract, ctx, getWrites } = fixture(AppleContract).registeredUsers(user).savedState(tree);
 
-  const dto = new PickAppleDto(tree.plantedBy, tree.variety, tree.index).signed(user.privateKey);
+  const dto = new PickAppleDto(tree.plantedBy, tree.variety, tree.index, randomUniqueKey()).signed(
+    user.privateKey
+  );
```

Comment:
minor syntax preference, optional suggestion

```suggestion
  const dto = new PickAppleDto(
    tree.plantedBy,
    tree.variety,
    tree.index,
    randomUniqueKey()
  ).signed(
    user.privateKey
  );
```